### PR TITLE
<Fix> evaluation dataset, printed samples

### PR DESCRIPTION
**Pull request by hypothe**
Bunch of minor "theoretical" changes in the evaluation function:

1. `test_data_gen` was used as the data generator in the evaluation, instead of `data_generator`, thereby evaluating the net on the test set used for training (not an actual issue here given the sequences are randomized and not sampled from existing datasets, but in principle would lead to a data leak in realistic scenarios);
2. the `correct` sequences printed were a sampling (with reinsertion) of the first 10 evaluated, instead of 10 sampled from the whole set of correct ones;
3. the condition for printing the incorrectly classified sequences would declare the absence of misclassifications if `verbose==False`, independently of their actual presence;

### CODE REVIEW

File: 08-seq_classification.ipynb [hunk 1]
```diff
@@ -497,7 +497,7 @@
     "\n",
     "    with torch.no_grad():\n",
     "        for batch_idx in range(len(data_generator)):\n",
-    "            data, target = test_data_gen[batch_idx]\n",
+    "            data, target = data_generator[batch_idx]\n",
```
[the remark this answers is not in the document]
Comment (hypothe) [reply 08-seq_classification.ipynb:500]: Then why are we declaring the "data_generator", with a different seed, and not using it to sample data?
